File: typescript/test/e2e/advertisement.test.ts
import { v4 as uuid } from "uuid"
import { FrameworkRequest, Method } from "../../src/framework/FrameworkRequest";
import SqliteConnection from "../../src/framework/database/SqliteConnection";
import { FrameworkServer } from "../../src/framework/FrameworkServer";
import { SqliteConnectionFactory } from "../../src/framework/database/SqliteConnectionFactory";
import { DatabaseConnection } from "../../src/framework/database/DatabaseConnection";

let server: FrameworkServer
let connection: DatabaseConnection;
describe("Advertisement", () => {


    beforeAll(async () => {
        server = await FrameworkServer.start();
        connection = await SqliteConnectionFactory.createClient();
        await connection.execute('delete from advertisements;', []);
    })
    it("Should create a advertisement", async () => {

        const request = new FrameworkRequest(Method.POST, '/advertisement',
            { id: uuid(), description: 'Dream advertisement', password: 'myPassword' }
        )

        const actual = await server.route(request)

        const dbData = await connection.query("SELECT * FROM advertisements")

        expect(actual.statusCode).toBe(201);

        expect(dbData.length).toBe(1);

    });
});

File: kotlin/src/test/kotlin/e2e/AdvertisementTest.kt
package e2e

import framework.DependencyInjectionResolver
import framework.FrameworkRequest
import framework.Server
import framework.database.DatabaseConnection
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class AdvertisementTest {
    companion object {
        private const val DESCRIPTION = "Dream advertisement"
        private const val ID = "6fa00b21-2930-483e-b610-d6b0e5b19b29"
        private const val PASSWORD = "myPassword"
    }

    private lateinit var connection: DatabaseConnection

    @BeforeEach
    fun init() {
        this.connection = DependencyInjectionResolver().connection()
        this.connection.execute("DELETE FROM advertisements")
    }

    @Test
    fun `publish advertisement`() {

        val server = Server(DependencyInjectionResolver())

        val result = server.route(FrameworkRequest(
                FrameworkRequest.METHOD_POST,
                "advertisement",
                mapOf(
                    "id" to ID,
                    "description" to DESCRIPTION,
                    "password" to PASSWORD,
                )
            )
        )

        Assertions.assertEquals(201, result.statusCode)

        val resultSet = this.connection.query("SELECT * from advertisements;")
        var description = ""

        if (resultSet.next()) {
            description = resultSet.getString(2)
        }

        Assertions.assertEquals(DESCRIPTION, description)
    }
}
